### PR TITLE
Allow Minors to Change age_range on Source

### PR DESCRIPTION
**Pull request by cassidysymons**

### CODE REVIEW

File: microsetta_private_api/api/_consent.py [hunk 1]
```diff
@@ -62,37 +62,56 @@ def check_consent_signature(account_id, source_id, consent_type, token_info):
 def sign_consent_doc(account_id, source_id, consent_type, body, token_info):
     _validate_account_access(token_info, account_id)
 
+    human_consent_age_groups = ["0-6", "7-12", "13-17", "18-plus"]
```
Comment (wasade) [inline microsetta_private_api/api/_consent.py:65]: Are these defined as static strings somewhere, and can those existing variables be used?

File: microsetta_private_api/api/tests/test_api.py [hunk 2]
```diff
@@ -1757,30 +1766,146 @@ def sign_data_consent(self):
...
+            content_type='application/json',
+            data=json.dumps(consent_data),
+            headers=self.dummy_auth)
+
+        self.assertEquals(201, response.status_code)
 
-        consent_data = copy.deepcopy(DUMMY_HUMAN_SOURCE)
-        consent_data.update("consent_type", ADULT_BIOSPECIMEN_CONSENT)
-        consent_data.update("consent_id", CONSENT_ID_BIO)
+    def sign_data_consent_new_age_invalid(self):
```
Comment (wasade) [inline microsetta_private_api/api/tests/test_api.py:1799]: This method won't execute as a test unless prefixed with `test_`

```suggestion
    def test_sign_data_consent_new_age_invalid(self):
```

File: microsetta_private_api/api/tests/test_api.py [hunk 3]
```diff
@@ -1757,30 +1766,146 @@ def sign_data_consent(self):
     def sign_biospecimen_consent(self):
```
Comment (wasade) [inline microsetta_private_api/api/tests/test_api.py:1766]: ...same here

Are these actually running? Historically, the unit test module only executed methods prefixed with `test` as part of the test suite

File: microsetta_private_api/api/tests/test_api.py [hunk 4]
```diff
@@ -1757,30 +1766,146 @@ def sign_data_consent(self):
...
+            '/api/accounts/%s/sources/%s/consent/%s' %
+            (dummy_acct_id, dummy_source_resp["source_id"], DATA_CONSENT),
+            content_type='application/json',
+            data=json.dumps(consent_data),
+            headers=self.dummy_auth)
+
+        # And assert that it fails
+        self.assertEquals(403, response.status_code)
+
+    def sign_data_consent_new_age_valid(self):
```
Comment (wasade) [inline microsetta_private_api/api/tests/test_api.py:1854]: same

File: microsetta_private_api/repo/consent_repo.py [hunk 5]
```diff
@@ -135,12 +136,16 @@ def is_consent_required(self, source_id, age_range, consent_type):
             True if the user needs to reconsent, False otherwise
         """
         if age_range == "18-plus":
```
Comment (wasade) [inline microsetta_private_api/repo/consent_repo.py:138]: these strings are replicated a lot in the codebase...

File: microsetta_private_api/repo/consent_repo.py [hunk 6]
```diff
@@ -159,15 +164,20 @@ def is_consent_required(self, source_id, age_range, consent_type):
...
+            # consent_id column in ag.consent_audit, whereas kids and
+            # teens use the assent_id column.
+            sql = """SELECT ca.signature_id
+                     FROM ag.consent_audit ca
+                     INNER JOIN ag.consent_documents cd
+                     ON {0} = cd.consent_id
+                     WHERE cd.version = %s
+                     AND cd.consent_type = %s
+                     AND ca.source_id = %s"""
+            sql = sql.format(consent_join)
```
Comment (wasade) [inline microsetta_private_api/repo/consent_repo.py:177]: Can this use SQL string composition instead (https://www.psycopg.org/docs/sql.html#module-psycopg2.sql)? Risk of injection right now is low (non-existent?) with current code, but doing it now is defensive for future change

**Review by wasade — CHANGES_REQUESTED** [review]

**Comment by cassidysymons** [comment]
@wasade Ready for re-review, thanks!

**Review by wasade — APPROVED** [review]
Thank you, @cassidysymons!

**Pull request merged**